conjure-java-core/src/integrationInput/java/com/palantir/product/EmptyUnionTypeExample.java

```diff
@@ -31,6 +31,10 @@ private Base getValue() {
         return value;
     }
 
+    public static EmptyUnionTypeExample unknown(String type, Map<String, Object> value) {
```

Comment:
These should end up:
```suggestion
    public static EmptyUnionTypeExample unknown(String type, Object value) {
```

I believe the map is an implementation detail, it should always be a singleton in which the key is the value of `type`.

We should add some validation that `type` doesn't match any of the known types, otherwise it would be possible to create an invalid value for a known type

Reply:
Ah nice, so you are imagining:
- Change param to `Object`
- In the body:
  - add a Preconditions check that the type is not known
  - populate the map when calling the UnknownWrapper constructor

Reply:
Yep, exactly :-)